src/Event.tsx

```diff
@@ -28,20 +28,37 @@ interface EventProps {
 }
 
 interface EventState {
-  event?: core.Event
+  event?: core.Event,
+  grayedOut: boolean
```

Comment:
Can we name this variable something more meaningful? If I'm looking at this code for the first time with no foreknowledge of the issue, I don't know _what_ is grayed out. Maybe change it to something like `userHasLogAccess` (and default it to `false`).

Reply:
Ohh Okk for sure. Kindly just confirm if the logic is correct. I will change the variable names accordingly.

Reply:
The other comments should shed light on that.

Reply:
Done ! Will be there in the next commit.